CMSgov/design-system
# [NO-TICKET] New, interactive release script

## Summary

- Rewrites our release script in TypeScript
- It now interactively guides us through the release process and includes bumping the main branch and drafting release notes
- Upgrade Lerna from v3.20 to v5.6

## How to test

1. Check out the `test/release-branch` branch, which includes these changes plus some fake changes to the core package so that lerna will allow a version bump
2. Run `yarn release`. Go ahead and do everything it says. You can choose to draft the release notes but not publish the draft in the `Do these notes look OK to publish as a Draft? (Y/n):` step.
3. After that, undo the release with `yarn release --undo`

Note that [it automatically created this pr when I ran it](https://github.com/CMSgov/design-system/pull/2655)
